markdown/events/2021/bytesize-3-code-structure.md

```diff
@@ -30,3 +30,109 @@ She will talk through the different files found in the nf-core template and desc
+
+We also have the license file, which describes the MIT license that all of the nf-core pipelines adhere to or apply, and a readme markdown file.
+
+The readme markdown file contains the easiest pipeline recommendations, such as how to get started to launch the pipeline.
+
+What are pipelines actually? They enable the execution of different tools one after another, so it's pretty important for any pipeline to have proper software packaging possibilities. This is natively supported by Nextflow; Nextflow natively supports the execution of pipelines using many different container engines and also Conda.
+
+We support this also of course in all of the nf-core pipelines.
+
+So how is software packaging defined?
+
+As part of the nf-core pipelines, we have mainly two files: the Dockerfile and the environment.yml. The environment.yml is a Conda environment file that lists all
+of the software dependencies of the pipeline and their versions. It can be used to just execute the pipeline by using Conda.
+
+However due to reproducibility issues that the conda environment can sometimes have, we recommend actually running the pipeline in a containerized fashion instead. This can be done mainly with Docker containers or Singularity containers. All pipelines support this, and there are other container engines that will also be supported.
+
+And what is the Docker container like?
+
+So for all the pipelines, there is one Docker container available that contains all the dependencies, and I'm going to show you how this container is defined in the Docker file. We have pretty minimalistic Docker files in the nf-core pipelines, so the docker file takes a container image that is directly provided by nf-core tools. What this does is that it copies this environment.yml file there, and creates a Conda environment inside of the container that adds this environment to the path.
+
+So you can see that all the tools for running the pipeline in the containerized fashion are also defined in this Conda environment file. This adds a pretty strict requirement of all of the nf-core pipelines; all of the software is available in Conda, so - at any of the Conda public channels. We have two main directories here as part of the pipeline template and one directory is called /bin, and this contains executable scripts that are used as part of the pipeline and can be written in any language. Here for example, it is in Python. Adding scripts to /bin provides the additional advantage of Nextflow adding all of these scripts to the path directly. So when you want to call descriptions as part of your pipeline, you can call the script directly and Nextflow will find it inside the bin.
+
+Additionally, we have another directory that's called /assets, and this directory contains the different templates that are used to send emails at the end of the pipeline saying what the status of the execution is: why the run passed or the run failed etc.
```

Comment:
```suggestion
Additionally, we have another directory that's called `/assets`, and this directory contains the different templates that are used to send emails at the end of the pipeline saying what the status of the execution is: why the run passed or the run failed etc.
```

If you want to use the file-path notation better to put it in code

Reply:
OK, didn't know how to do that. Thanks!